packages/host/app/components/operator-mode/container.gts

```diff
@@ -590,9 +607,60 @@ export default class OperatorModeContainer extends Component<Signature> {
   };
 
   @action updateSubmode(submode: Submode) {
+    switch (submode) {
+      case Submode.Interact:
+        this.onCodePathChange(null);
+        break;
+      case Submode.Code:
+        this.onCodePathChange(new URL(this.lastCardInRightMostStack.id));
+        break;
+    }
+
     this.operatorModeStateService.updateSubmode(submode);
   }
 
+  @action onCodePathChange(codePath: URL | null) {
+    this.operatorModeStateService.updateCodePath(codePath);
+    this.loadCardForCodeMode.perform(codePath);
+  }
+
+  loadCardForCodeMode = restartableTask(async (codePath: URL | null) => {
+    try {
+      if (!codePath) return;
+
+      const realmsToCheck = [
+        this.cardService.defaultURL.href,
+        baseRealm.url,
+        ...otherRealmURLs,
+      ];
+      const isInAnyRealms = await Promise.all(
+        realmsToCheck.map(async (realmURL) => {
+          const realmPath = new RealmPaths(realmURL);
+          return realmPath.inRealm(codePath);
+        }),
+      );
```

Comment:
let's turn this into a method that the card-service can provide. There are other cases where there is a need to see if a URL is in a particular realm.

it could be: `CardService.getRealmForURL(url: URL): URL | undefined` where you give it a URL and the method will return the realm that contains the given URL or undefined if it matches no realms

Reply:
actually i just added this for an unrelated feature that also needed this functionality. if you merge upstream changes you'll get this: https://github.com/cardstack/boxel/blob/d0a5999b4ae29c9d9363a91fe421e6a2a0c7350b/packages/host/app/services/card-service.ts#L324-L332